numba/cext/listobject.c

```diff
@@ -441,6 +399,50 @@ numba_list_resize(NB_List *lp, Py_ssize_t newsize) {
     return LIST_OK;
 }
 
+/* Delete a single item.
+ *
+ * lp: a list
+ * index: the index of the item to delete
+ *
+ * */
+int
+numba_list_delitem(NB_List *lp, Py_ssize_t index) {
```

Comment:
The index is restricted to C semantics here, i.e. non-negative? If so perhaps add to docstring.

Reply:
The `index` appears in many of the functions, so for consistency it should probably be set everywhere. How about something like:

```
 * index: the index of the item to delete (must be in range 0 <= index < len(list))
```

Reply:
Sounds good.

Reply:
updated!